src/main/java/lotto/domain/LottoNumber.java

```diff
@@ -0,0 +1,44 @@
+package lotto.domain;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import javax.swing.text.html.Option;
+
+public class LottoNumber {
+	private List<Integer> lottoNumber;
+
+	public LottoNumber() {
+	}
+
+	public LottoNumber(List<Integer> lottoNumber) {
+		this.lottoNumber = lottoNumber;
+	}
+
+	public List<Integer> getLottoNumber() {
+		return lottoNumber;
+	}
+
+	public void generate(NumberGenerator numberGenerator) {
+		this.lottoNumber = numberGenerator.generateNumbers();
+	}
+
+	private int countSameNumber(LottoNumber winLottoNumber) {
+		return (int) lottoNumber.stream()
+			.filter(ln -> winLottoNumber.getLottoNumber().contains(ln))
+			.count();
+	}
+
+	public Optional<LottoRank> rankLotto(LottoNumber winLottoNumber) {
```

Comment:
```suggestion
	public LottoRank rankLotto(LottoNumber winLottoNumber) {
		int sameNumberCount = countSameNumber(winLottoNumber);
		return LottoRank.valueOfCount(sameNumberCount);
	}
```

1. Optional이 필요한 상황이 아닙니다. 리턴 타입을 optional로 두면 클라이언트 코드가 복잡해집니다.
2. LottoRank 객체에서 값을 꺼내서 찾지 말고 직접 물어보는게 좋겠네요

Reply:
findFirst할 때 Optional로 반환해서 그대로 return 값으로 전달을 하느라 그랬는데,,
.findFirst().orElse(LottoRank.MISS); 
이렇게 변경하였습니다.

Reply:
필요하다면 orElseThrow로 예외처리 하셔도 됩니다